packages/teams-js/src/public/runtime.ts

```diff
@@ -7,9 +7,25 @@ import { HostClientType } from './constants';
 
 const runtimeLogger = getLogger('runtime');
 
-export interface IRuntime {
+export interface IBaseRuntime {
   readonly apiVersion: number;
   readonly isLegacyTeams?: boolean;
+}
+
+/**
+ * Latest runtime interface version
+ */
+export type Runtime = IRuntimeV1;
+
+export const latestRuntimeApiVersion = 1;
+
+function isLatestRuntimeVersion(runtime: IBaseRuntime): runtime is Runtime {
+  return runtime.apiVersion === latestRuntimeApiVersion && isRuntimeV1(runtime);
```

Comment:
This is something that came up when I was testing the upgrade mechanism. I wrote an IUpgrade that did a v0 to v1 transformation, and I needed an isV1 check so that the `upgrade` function could return a v1 object. I thought it might be good to just codify writing type checks as part of the upgrade process and then using them in this function, but maybe it would be better to leave that up to developers implementing upgrades.

Reply:
The other thing I was trying to do was tie the `latestRuntimeApiVersion` to the upgrade process. `isLatestRuntimeVersion` is used to verify that the runtime we have at the end of the upgrade chain is the version we want, so using `latestRuntimeApiVersion` here makes sure that number is always in sync with the Runtime interface, and then isRuntimeV1 would actually make sure the object is well-formed and has all required properties.

Reply:
I consolidated this to a single type check. Looks like we can do the version transformations without explicit type checks so I'll leave that up to whoever's implementing them. This makes the upgrade process a little more straightforward as well